File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int leftsum=0;
        int rightsum=0;
        int maxsum=0;
        for(int i=0;i<k;i++)
        {
            leftsum=leftsum+cardPoints[i];
        
        }
        maxsum=leftsum;
        int rightlen=cardPoints.size()-1;
        for(int i=k-1;i>=0;i--)
        {
           leftsum=leftsum-cardPoints[i];
           rightsum=rightsum+cardPoints[rightlen];
           rightlen=rightlen-1;
           
            maxsum=max(maxsum,leftsum+rightsum);
        }
         return maxsum;
        
    }
   
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int n=cardPoints.length;
        int lsum=0;
        int rsum=0;
        int sum=0;
        for(int i=0;i<=k-1;i++)
        {
            lsum+=cardPoints[i];
            sum=lsum;
        }
        int ridx=n-1;
        for(int i=k-1;i>=0;i--)
        {
            lsum=lsum-cardPoints[i];
            rsum+=cardPoints[ridx];
            ridx=ridx-1;
            sum=Math.max(sum,lsum+rsum);
        }

        return sum;
        
    }
}